Buildings/Fluid/SolarCollectors/BaseClasses/ASHRAESolarGain.mo

```diff
@@ -115,8 +119,9 @@ equation
   for i in 1 : nSeg loop
     QSol_flow[i] = A_c/nSeg*(y_intercept*iam*(HDirTil*(1.0 -
     shaCoe_internal) + HSkyDifTil + HGroDifTil))*
-    Buildings.Utilities.Math.Functions.smoothHeaviside(
-     (Medium.T_max+1)-TFlu[i],1);
+      smooth(1, if TFlu[i] < TMedMax
+        then 1
+        else Buildings.Utilities.Math.Functions.smoothHeaviside(TMedMax-TFlu[i], dTMax));
```

Comment:
This code is discontinuous:
```
model test "test"
  Real x;
  parameter Real TMedMax=1;
  parameter Real dTMax=1;

equation 
x=smooth(1, if time < TMedMax
        then 1
        else Buildings.Utilities.Math.Functions.smoothHeaviside(TMedMax-time, dTMax));
  annotation (uses(Modelica(version="3.2.1")));
end test;
```
![screen shot 2015-06-30 at 08 13 26](https://cloud.githubusercontent.com/assets/5970879/8424848/f0a82ae6-1eff-11e5-8c8c-9e94fc2d19db.png)